overlord/hookstate/context.go

```diff
@@ -0,0 +1,111 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package hookstate
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/snapcore/snapd/overlord/state"
+	"github.com/snapcore/snapd/snap"
+)
+
+const (
+	contextDataKey = "hook-context"
+)
+
+// Context represents the context under which a given hook is running.
+type Context struct {
+	task    *state.Task
+	setup   hookSetup
+	dataKey string
+}
+
+func newContext(task *state.Task, setup hookSetup) *Context {
+	return &Context{
+		task:    task,
+		setup:   setup,
+		dataKey: setup.Hook + "-" + contextDataKey,
```

Comment:
I think this can be "hook-context", hardcoded. We shouldn't have more than a hook per task, otherwise we compromise uniqueness.